### PR TITLE
Document s390x-unknown-linux targets

### PR DESCRIPTION
**Pull request by uweigand**
This adds documentation for the following existing targets:
  s390x-unknown-linux-gnu (Tier 2 with host tools)
  s390x-unknown-linux-musl (Tier 3)

I volunteer as maintainer for these targets going forward.

### CODE REVIEW

**Comment by ehuss** [comment]
@wesleywiser or @davidtwco, is there any process for adding new maintainers? Just add anyone that volunteers?

cc @cuviper since I've seen you touch s390 stuff before.

**Comment by uweigand** [comment]
> @wesleywiser or @davidtwco, is there any process for adding new maintainers? Just add anyone that volunteers?

Well, that's what I was told to do here: https://github.com/rust-lang/rust/issues/113739#issuecomment-2448390097 ... If there's any other process, I'd be happy to follow that as well.

As additional context, I did actually write most of the initial s390x support code back in the day: https://github.com/rust-lang/rust/pull/36369; I'm also the LLVM back-end code owner for the platform.

**Comment by cuviper** [comment]
I'm interested in the GNU target insofar as it's relevant to Fedora and RHEL, and I can be an additional contact if needed, but @uweigand is definitely the better expert on this architecture.

**Review by wesleywiser — APPROVED** [review]

**Comment by uweigand** [comment]
> I'm interested in the GNU target insofar as it's relevant to Fedora and RHEL, and I can be an additional contact if needed, but @uweigand is definitely the better expert on this architecture.

Hi @cuviper, of course I'd be happy to add you to the list if you volunteer!

**Comment by wesleywiser** [comment]
@ehuss if there are existing target maintainers, we cc them on proposed new maintainers and see what they think. In this case, since there's no existing official target maintainers, the main thing we just look at is contribution history. I see @uweigand has a long history of supporting this target so this all looks good to me.

Happy to r+ when the question of adding @cuviper as another maintainer is settled!

File: src/doc/rustc/src/platform-support/s390x-unknown-linux-gnu.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# `s390x-unknown-linux-gnu`
+
+**Tier: 2 (with Host Tools)**
+
+IBM z/Architecture (s390x) targets (including IBM Z and LinuxONE) running Linux.
+
+## Target maintainers
+
+- Ulrich Weigand, <ulrich.weigand@de.ibm.com>, [@uweigand](https://github.com/uweigand)
```
Comment (cuviper) [inline src/doc/rustc/src/platform-support/s390x-unknown-linux-gnu.md:9]:
```suggestion
- Ulrich Weigand, <ulrich.weigand@de.ibm.com>, [@uweigand](https://github.com/uweigand)
- Josh Stone, <jistone@redhat.com>, [@cuviper](https://github.com/cuviper)
```

Comment (uweigand) [reply src/doc/rustc/src/platform-support/s390x-unknown-linux-gnu.md:9]: Thanks!

**Review by workingjubilee — COMMENTED** [review]
Thanks!

Yeah I left the two additional criteria implicit since they felt obviously fulfilled in this case. The null set is very welcoming and the maintainer obviously has the background and commitment.

**Pull request merged**